src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -518,33 +518,37 @@ public static int findRepeatedSubstring(byte[] bases) {
      * @return                       Number of repetitions (0 if testString is not a concatenation of n repeatUnit's
      */
     public static int findNumberOfRepetitions(byte[] repeatUnit, byte[] testString, boolean lookForward) {
-        int numRepeats = 0;
+        final int repeatLength = repeatUnit.length;
+        final int testLength = testString.length;
+        final int lengthDifference = testLength - repeatLength;
+
         if (lookForward) {
+            int numRepeats = 0;
             // look forward on the test string
-            for (int start = 0; start < testString.length; start += repeatUnit.length) {
-                int end = start + repeatUnit.length;
-                byte[] unit = Arrays.copyOfRange(testString,start, end);
-                if(Arrays.equals(unit,repeatUnit))
+            for (int start = 0; start < testLength; start += repeatLength) {
+                if(start <= lengthDifference && Utils.equalRange(testString, start, repeatUnit, 0, repeatLength)) {
```

Comment:
repeatLength == 0 can only be tolerated if testString has length 0. This is still a bogus condition so it'll disallow it (throw) and add tests for it,